### DISCUSSION TITLE
Running the app in background or on lock screen

### DISCUSSION THREAD

**Discussion by i-mage** [quasarframework/quasar#11682] [CLI - Capacitor / Cordova modes]
Is it possible to keep running the app in background? My application is playing music and it stops immediately once the screen is locked.

**Comment by hawkeye64** [comment]
@i-mage Unfortunately, no, not on a locked screen. It is possible on mobile, but you have to set up the policies and provide a separate UI for locked screen controls (pause, play, etc., like Spotify does).

Otherwise, it is up to the User. You can find more info here: https://www.alphr.com/play-youtube-with-phone-locked/

**Reply by i-mage** [reply]
So basically Quasar applicatoin cannot run in the background while switching to home screen? I don't need to control it just keep it running. 

I have found this but yet didn't have time to implement it:

https://github.com/robingenz/capacitor-background-task
https://capacitorjs.com/docs/v2/apis/background-task

**Reply by hawkeye64** [reply]
With those, I think you get 30 more seconds before your app is completely put to sleep.

**Reply by i-mage** [reply]
Thanks for the insight. So it is possible to develop background running apps only in Swift (for iOS)?

**Reply by hawkeye64** [reply]
I believe so. But, I haven't made a mobile app in years and things change rapidly in this space. But I don't think hybrid apps are allowed certain privileges but native apps are.
